### PR TITLE
Drop py2 support; Extend py3 support; Fix CI; enable dependabot

### PR DESCRIPTION
**Pull request by MatthijsBurgh**

### CODE REVIEW

**Comment by MatthijsBurgh** [comment]
Ping @gruns

**Comment by MatthijsBurgh** [comment]
Friendly ping @gruns ;)

**Comment by gruns** [comment]
hey @MatthijsBurgh 

1) thank you for the ping! *always* helpful for a bump
2) THANK YOU FOR YOUR PR! 🙏. PRs are the best. thank you
3) zooming out, are there any exigent reasons to drop py2? i am *all* for simplifying things. for dropping baggage. lean and mean. but curious for your thoughts here

thank you! 🙌

**Comment by a-detiste** [comment]
assertRegexpMatches is gone from Py3.13

**Comment by gruns** [comment]
took a quick look: icecream's py27 downloads are about 0.4%. see https://pypistats.org/packages/icecream

0.4% is absolutely low enough to be worth deprecating if it makes future work and maintenance easier. people who need py2 support can download old versions of icecream

**Comment by gruns** [comment]
> assertRegexpMatches is gone from Py3.13

@a-detiste: great eyeballs. a quick grep only shows one instance of `assertRegexpMatches()` in the existing HEAD (ie not this PR)

```
% grep -riI 'assertRegexpMatches' --exclude-dir='./.*' .
./tests/test_icecream.py:        self.assertRegexpMatches(
```

File: tests/test_icecream.py [hunk 1]
```diff
@@ -600,7 +588,7 @@ def testMultilineContainerArgs(self):
                     list(range(15))])
 
         lines = err.getvalue().strip().splitlines()
-        self.assertRegexpMatches(
+        self.assertRegex(
             lines[0],
```
Comment (a-detiste) [inline tests/test_icecream.py:592]: Here ?

Comment (MatthijsBurgh) [reply tests/test_icecream.py:592]: assertRegexpMatches is been replaced by assertRegex. So this is future proof

**Comment by gruns** [comment]
merged! 🎉 

@MatthijsBurgh everyone thanks you, again, for this wonderful PR. wonderful work. thank you. 🙏

**Pull request merged**